workspaces/ui-v2/src/pages/docs/components/SimulatedBodyPreview.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { useEffect, useMemo, useState } from 'react';
+
+import { CQRSCommand } from '@useoptic/optic-domain';
+
+import { ShapeFetcher, SimulatedCommandStore } from '<src>/components';
+import { useSpectacleContext } from '<src>/contexts/spectacle-provider';
+import { useAppSelector } from '<src>/store';
+import { IShapeRenderer } from '<src>/types';
+import { SpectacleClient } from '<src>/clients';
+
+type SimulatedBodyProps = {
+  rootShapeId: string;
+  endpointId: string;
+  children: (shapes: IShapeRenderer[]) => React.ReactElement;
+};
+
+export const SimulatedBody = ({
+  rootShapeId,
+  endpointId,
+  children,
+}: SimulatedBodyProps) => {
+  const spectacle = useSpectacleContext();
+  const spectacleClient = useMemo(() => new SpectacleClient(spectacle), [
+    spectacle,
+  ]);
+  const [previewCommands, setPreviewCommands] = useState<CQRSCommand[]>([]);
+  // Commands that generate commands - removed fields
+  const removedFields = useAppSelector(
+    (state) => state.documentationEdits.fieldEdits.removedFields
+  );
+
+  useEffect(() => {
+    let isStale = false;
+    (async () => {
+      // TODO FLEB add in edited fields
+      const removeFieldCommandsPromise: Promise<CQRSCommand[]> = Promise.all(
+        removedFields.map((fieldId) =>
+          spectacleClient.fetchFieldRemoveCommands(fieldId)
+        )
+      ).then((fieldCommands) => fieldCommands.flat());
```

Comment:
We call this twice (command preview generation, and in the save documentation changes), which is not ideal, but it is simpler to implement. Otherwise we would have to add in async state into redux, check for loading states and only allow submitting this was "loaded".

Since this is fairly low overhead and only 1 extra call, I think this is acceptable until we see we want to invest in performance